server/service/apple_mdm.go

```diff
@@ -3438,6 +3471,249 @@ func ReconcileAppleProfiles(
 	return nil
 }
 
+func preprocessProfileContents(
+	ctx context.Context,
+	appConfig *fleet.AppConfig,
+	ds fleet.Datastore,
+	targets map[string]*cmdTarget,
+	profileContents map[string]mobileconfig.Mobileconfig,
+) error {
+
+	// This method replaces Fleet variables ($FLEET_VAR_<NAME>) in the profile contents, generating a unique profile for each host.
+	// For a 2KB profile and 30K hosts, this method may generate ~60MB of profile data in memory.
+
+	isNDESSCEPConfigured := func(profUUID string, target *cmdTarget) (bool, error) {
+		if !license.IsPremium(ctx) {
+			for _, hostUUID := range target.hostUUIDs {
+				err := ds.UpdateOrDeleteHostMDMAppleProfile(ctx, &fleet.HostMDMAppleProfile{
```

Comment:
the red flag I see here is that we'll potentially perform this call 30k times in a row. In the reconciler we've been trying to do things in bulk. Could we use `ds.BulkUpsertMDMAppleHostProfiles`?

same applies for all other times we do something similar on this method.